biojava-core/src/main/java/org/biojava/nbio/core/search/io/Hsp.java

```diff
@@ -0,0 +1,251 @@
+    private Integer mismatchCount = null;
+    private SimpleSequencePair<S, C> returnAln;
+
+    @Override
+    public int hashCode() {
+        int hash = 5;
+        hash = 67 * hash + (this.hspQseq != null ? this.hspQseq.hashCode() : 0);
+        hash = 67 * hash + (this.hspHseq != null ? this.hspHseq.hashCode() : 0);
+        hash = 67 * hash + (this.hspIdentityString != null ? this.hspIdentityString.hashCode() : 0);
+        return hash;
+    }
+    /**
+     * Experimental.
+     * Wants to implement conceptual comparisons of search results.
+     * Fields unrelated to search are deliberately not considered.
+     * 
+     * In HSP case, alignment representation strings are considered.
+     * @return true if HSP alignments are the same, 
+     * false otherwise or if alignment strings are undetermined
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final Hsp<?, ?> other = (Hsp<?, ?>) obj;
+        if ((this.hspQseq == null) ? (other.hspQseq != null) : !this.hspQseq.equals(other.hspQseq)) {
+            return false;
+        }
+        if ((this.hspHseq == null) ? (other.hspHseq != null) : !this.hspHseq.equals(other.hspHseq)) {
+            return false;
+        }
+        if ((this.hspIdentityString == null) ? (other.hspIdentityString != null) : !this.hspIdentityString.equals(other.hspIdentityString)) {
+            return false;
+        }
+        return true;
+    }
+    
+    public SequencePair<S,C> getAlignment(){
+        if (returnAln != null) return returnAln;
+        
+        SimpleAlignedSequence alignedQuery, alignedHit;
+        // queryFrom e hitTo?
+        int numBefore, numAfter;
+        
+        alignedQuery = new SimpleAlignedSequence(getSequence(hspQseq), getAlignmentsSteps(hspQseq));
+        alignedHit = new SimpleAlignedSequence(getSequence(hspHseq), getAlignmentsSteps(hspHseq));
+        
+        returnAln = new SimpleSequencePair<S, C>(alignedQuery, alignedHit);
+        
+        return returnAln;
+    }
+    
+    private Sequence getSequence(String gappedSequenceString){
+        if (gappedSequenceString == null) return null;
+        
+        Sequence returnSeq = null;
+        String sequenceString = gappedSequenceString.replace("-", "");
+        
+        try {
+            if (sequenceString.matches("^[ACTG]+$")) 
+                returnSeq = new DNASequence(sequenceString, DNACompoundSet.getDNACompoundSet());
+            else if (sequenceString.matches("^[ACUG]+$"))
+                returnSeq = new RNASequence(sequenceString, DNACompoundSet.getDNACompoundSet());
+            else
+                returnSeq = new ProteinSequence(sequenceString, AminoAcidCompoundSet.getAminoAcidCompoundSet());
+        } catch (CompoundNotFoundException ex) {
+            Logger.getLogger(Hsp.class.getName()).log(Level.SEVERE, null, ex);

```

Comment:
Logging in Biojava (since Biojava 4) is happening through the slf4j interface. You should change this to use slf4j. You simply need to import on top:
```java
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
```
then declare a static variable for the logger:
```java
private static final Logger logger = LoggerFactory.getLogger(Hsp.class);
```
and then use it like:
```java
logger.error("Unexpected error, could not find compound blah blah ", ex);
```
See for instance `org.biojava.nbio.structure.StructureImpl` for an example of the above.